pkg/u8g2/src/csrc/u8x8_riotos.c

```diff
@@ -146,20 +168,29 @@ uint8_t u8x8_byte_riotos_hw_spi(u8x8_t *u8g2, uint8_t msg, uint8_t arg_int, void
 
     return 1;
 }
-#endif /* SPI_NUMOF */
+#endif /* MODULE_PERIPH_SPI */
 
-#ifdef I2C_NUMOF
+#ifdef MODULE_PERIPH_I2C
 uint8_t u8x8_byte_riotos_hw_i2c(u8x8_t *u8g2, uint8_t msg, uint8_t arg_int, void *arg_ptr)
 {
-    static uint8_t buffer[255];
-    static uint8_t index;
+    static uint8_t buffer[32]; /* u8g2/u8x8 will never send more than 32 bytes between START_TRANSFER and END_TRANSFER */
+    static size_t index;
+
+    const u8x8_riotos_t u8x8_riot_ptr = u8g2->user_ptr;
+
+    // Check that user_ptr is correctly set
+    if (NULL == u8x8_riot_ptr) {
+        LOG_ERROR("u8x8_riot_ptr cannot be NULL");
+        return 1;
+    }
 
-    i2c_t dev = (i2c_t) u8g2->dev;
+    i2c_t dev = I2C_DEV(u8x8_riot_ptr->device_index);
```

Comment:
In most drivers, the I2C_DEV define is moved to where `u8x8_riot_ptr->device_index` is set.

Reply:
I did not like the fact, that I would store a value of type i2c_t or spi_t in int, therefore I assumed this to be more clear solution, however, if you feel this is not suitable, I can change it - what type to use then? plain int?

Reply:
I found another way (sorry), but this aligns the implementation with other drivers.

https://github.com/RIOT-OS/RIOT/blob/master/tests/driver_enc28j60/Makefile#L26

So remove the `I2C_DEV(..)/SPI_DEV(..)` from the source code, and move it to the Makefile.

Reply:
Sorry, I do not understand exactly how and why to do so. The code you are referring is for the test, but the original code is from the actual package. If I somehow select the bus (SPI or I2C) in makefile, then I would remove the possibility to have two displays, one on SPI and one on I2C at the same time - and that is something I would like to avoid. 

In my proposal of the implementation, the user is simply selecting the bus by selecting the callback, and in the provided structure, they only need to provide the "index" of the selected bus.

One solution which comes to my mind is to provide both i2c_t and spi_t variables in the user structure, and the user would have to fill in the one corresponding to the desired bus - would that be a suitable option?

Reply:
If you do it like in that test Makefile, the `TEST_SPI_DEV = SPI_DEV(..)` ends up in `device_index`, so it is correct. Normally, the end-user would do the same, but from C code.